### PR TITLE
DEVX-1719 adds documentation to docker-compose files

### PR DESCRIPTION
**Pull request by rspurgeon**
explaining our usage of environment variables throughout.

### CODE REVIEW

File: ccloud/beginner-cloud/docker-compose.yml [hunk 1]
```diff
@@ -1,3 +1,15 @@
+# docker-compose supports environment variable substitution with the ${VARIABLE-NAME} syntax.
+# Environment variables can be sourced in a variety of ways.  One of those ways is through
+# a well known '.env' file located next to the docker-compose.yml file.  See the Docker
```
Comment (ybyzek) [inline ccloud/beginner-cloud/docker-compose.yml]:
```suggestion
# a well known '.env' file located in the same folder as the docker-compose.yml file.  See the Docker
```

File: ccloud/beginner-cloud/docker-compose.yml [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+# documentation for details: https://docs.docker.com/compose/environment-variables/#the-env-file
+# 
+# This feature is being used to parameterize some values within this file.  In this directory is also
+# a .env file, which is actually a symbolic link to <examples-root>/utils/config.env.  That file
+# contains values which get substituted here when docker-compose parses this file.
+#
+# If you'd like to view the docker-compose.yml file rendered with it's environment variable substituions
```
Comment (ybyzek) [inline ccloud/beginner-cloud/docker-compose.yml]:
```suggestion
# If you'd like to view the docker-compose.yml file rendered with its environment variable substituions
```

File: ccloud/beginner-cloud/docker-compose.yml (hunk 1 above)
Comment (rspurgeon) [reply ccloud/beginner-cloud/docker-compose.yml]: resolved outside of GH suggestions...

File: ccloud/beginner-cloud/docker-compose.yml (hunk 2 above)
Comment (rspurgeon) [reply ccloud/beginner-cloud/docker-compose.yml]: resolved, thank you.  i should have paid better attention in grammar school

**Review by ybyzek — APPROVED** [review]
LGTM @rspurgeon

**Pull request merged**